### PR TITLE
feat: Notify When Group Join Fails

### PR DESCRIPTION
**Pull request by alexrisch**
When silent handling of group join requests failed then notify the user

Closes https://github.com/ephemeraHQ/converse-app/issues/982

### CODE REVIEW

**Review by nmalzieu — APPROVED** [review]
code looks clean
@alexrisch what is it handling exactly?
- it shows a notification to the admin if failure to add to a group? what is the content of the notification? is there a deeplink to the group page?
- do we know when this usually fails? did we ever log it? or you suspect this is issue we have when Joining... is too long?
- should there be an Android part as well?

**Comment by alexrisch** [comment]
@nmalzieu It's a good idea to add the deep link to the group, it helps the user accept easier, so will convert to draft to add that

File: screens/GroupInvite.tsx [hunk 1]
```diff
@@ -100,24 +92,20 @@ export default function GroupInviteScreen({
...
+    const groupsBeforeJoining = await fetchGroupsQuery(account);
+    if (groupId && groupsBeforeJoining.byId[getTopicFromGroupId(groupId)]) {
+      // User has already been added to the group
+      handleNewGroup(groupsBeforeJoining.byId[getTopicFromGroupId(groupId)]);
+      return;
+    }
     logger.debug(
       `[GroupInvite] Before joining, group count = ${groupsBeforeJoining.ids.length}`
     );
-    let joinRequestId = getInviteJoinRequest(account, groupInvite?.id);
```
Comment (technoplato) [inline screens/GroupInvite.tsx:109]: was using an old `inviteJoinRequest` perhaps the culprit for people's joins timing out do you think?

if not - why was this removed in favor of always creating one?

Comment (alexrisch) [reply screens/GroupInvite.tsx:109]: 2 fold:
1. If a user presses again it will actually send the notification again, so has a chance to potentially reduce the manual step
2. Moved this check to the backend relevant PR is here: https://github.com/ephemeraHQ/converse-backend/pull/39
